connectors/httpclient/src/main/java/org/openstack4j/connectors/httpclient/HttpResponseImpl.java

```diff
@@ -123,8 +125,13 @@ public String header(String name) {
     @Override
     public <T> T readEntity(Class<T> typeToReadAs) {
         HttpEntity entity = response.getEntity();
+        if (entity == null) {
```

Comment:
Because `checkNotNull(entity)` would throw a `NullPointerException` if `entity == null`, which is exactly the issue #825. The goal of this pull request is to accept without error the case where `entity == null`, e.g. for HEAD requests.

For example, in Jersey2 connector, `readEntity` will return `null` for a HEAD request.